internal/service/resource_pool_object_handler.go

```diff
@@ -0,0 +1,205 @@
+// server. This is mandatory.
+func (b *ResourcePoolObjectHandlerBuilder) SetBackendURL(
+	value string) *ResourcePoolObjectHandlerBuilder {
+	b.backendURL = value
+	return b
+}
+
+// SetGraphqlQuery sets the query to send to the search API server.
+func (b *ResourcePoolObjectHandlerBuilder) SetGraphqlQuery(
+	value string) *ResourcePoolObjectHandlerBuilder {
+	b.graphqlQuery = value
+	return b
+}
+
+// SetGraphqlVars sets the query vars to send to the search API server.
+func (b *ResourcePoolObjectHandlerBuilder) SetGraphqlVars(
+	value *searchmodel.SearchInput) *ResourcePoolObjectHandlerBuilder {
+	b.graphqlVars = value
+	return b
+}
+
+// Build uses the data stored in the builder to create and configure a new handler.
+func (b *ResourcePoolObjectHandlerBuilder) Build() (
+	result *ResourcePoolObjectHandler, err error) {
+	// Check parameters:
+	if b.logger == nil {
+		err = errors.New("logger is mandatory")
+		return
+	}
+	if b.cloudID == "" {
+		err = errors.New("cloud identifier is mandatory")
+		return
+	}
+	if b.backendURL == "" {
+		err = errors.New("backend URL is mandatory")
+		return
+	}
+	if b.backendToken == "" {
+		err = errors.New("backend token is mandatory")
+		return
+	}
+
+	// Create the HTTP client that we will use to connect to the backend:
+	var backendTransport http.RoundTripper
+	backendTransport = &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+	if b.transportWrapper != nil {
+		backendTransport = b.transportWrapper(backendTransport)
+	}
+	backendClient := &http.Client{
+		Transport: backendTransport,
+	}
+
+	// Prepare the JSON iterator API:
+	jsonConfig := jsoniter.Config{
+		IndentionStep: 2,
+	}
+	jsonAPI := jsonConfig.Froze()
+
+	// Create and populate the object:
+	result = &ResourcePoolObjectHandler{
+		logger:        b.logger,
+		cloudID:       b.cloudID,
+		backendURL:    b.backendURL,
+		backendToken:  b.backendToken,
+		backendClient: backendClient,
+		jsonAPI:       jsonAPI,
+		graphqlQuery:  b.graphqlQuery,
+		graphqlVars:   b.graphqlVars,
+	}
+	return
+}
+
+// Get is part of the implementation of the collection handler interface.
+func (h *ResourcePoolObjectHandler) Get(ctx context.Context,
+	request *ObjectRequest) (response *ObjectResponse, err error) {
+	// Fetch the object:
+	resourcePool, err := h.fetchObject(ctx, request.ID)
+	if err != nil {
+		return
+	}
+
+	// Return the result:
+	response = &ObjectResponse{
+		Object: resourcePool,
+	}
+	return
+}
+
+func (h *ResourcePoolObjectHandler) fetchObject(ctx context.Context,
+	id string) (resourcePool data.Object, err error) {
+	// Filter results by 'cluster' property
+	h.graphqlVars.Filters = append(h.graphqlVars.Filters, &searchmodel.SearchFilter{
+		Property: "cluster",
+		Values:   []*string{&id},
+	})
+
+	// Fetch resource pools
+	resourcePools, err := NewResourcePoolFetcher(
+		h.logger, h.cloudID, h.backendURL, h.backendToken, h.graphqlQuery, h.graphqlVars,
+	).FetchItems(ctx)
```

Comment:
Is the resource pool fetcher stateless? Can we create one instance and reuse it for all calls to `fetchObject`?